torch/csrc/Device.cpp

```diff
@@ -65,9 +65,6 @@ PyObject *THPDevice_pynew(PyTypeObject *type, PyObject *args, PyObject *kwargs)
     int32_t device_index = -1;
     if (!r.isNone(1)) {
       device_index = r.toInt64(1);
-      // -1 is allowed in ATen/C++, to mean the default device, but not in
-      // Python.
-      TORCH_CHECK(device_index >= 0, "Device index must not be negative");
```

Comment:
What's going on here? This test doesn't seem harmful. Is this for lining up the C++ and Python behavior? A little bit of divergence is probably OK; it's better to allow less than to allow more (you can make things more relaxed in the future, but not vice versa)

Reply:
Yeah. I was a little torn because of exactly what you just said- taking it out now means we can't put it back later.

But I figured that the target state we want is to have consistent behavior between the python and C++. It seems a little weird to randomly throw an error in python and have the exact same logic pass in C++.

You think it's better to keep in for now?